crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -547,7 +562,11 @@ impl<'t> TextEdit<'t> {
             cursor_range = Some(new_cursor_range);
         }
 
-        let mut text_draw_pos = response.rect.min;
+        let mut text_draw_pos = match align {
+            Align::Min => response.rect.min,
+            Align::Center => response.rect.center_top() - Vec2::new(galley.size().x, 0.0) / 2.0,
+            Align::Max => response.rect.right_top() - Vec2::new(galley.size().x, 0.0),
+        };
```

Comment:
If you make `align` an `Align2`, then you can use `Align2::align_size_within_rect`